File: Programmin_in_C#-Erasmus/Final_Project/Validators/FriendshipValidator.cs
using Final_Project.Domain;
using Final_Project.Exceptions;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Final_Project.Validators
{
    public class FriendshipValidator : BaseValidator<Friendship>
    {
        public override void validate(Friendship frship)
        {
            if (frship.getIdUser1().Equals(frship.getIdUser2()))
            {
                throw new ValidationException("Invalid friendship");
            }
        }
    }
}

File: Programmin_in_Java-Erasmus/Final_Project/src/main/java/server/validators/FriendshipValidator.java
package main.java.server.validators;

import main.java.server.domain.Friendship;
import main.java.server.exceptions.ValidationException;

public class FriendshipValidator extends BaseValidator<Friendship> {
    /**
     * Checks if the friendship has two not null members
     * @param friendship the given frienship
     * @throws ValidationException if the friendship is invalid
     */
    @Override
    public void validate(Friendship friendship) {
        if (friendship.getIdUser1() == null || friendship.getIdUser2() == null ||
            friendship.getIdUser1() <= 0  || friendship.getIdUser2() <=0 ||
            friendship.getIdUser1().equals(friendship.getIdUser2()))
            throw new ValidationException("Invalid friendship");
    }
}
